provider/gce/google/conn_instance.go

```diff
@@ -90,16 +90,19 @@ func (gce *Connection) Instances(prefix string, statuses ...string) ([]Instance,
 func (gce *Connection) removeInstance(id, zone string) error {
 	err := gce.raw.RemoveInstance(gce.projectID, zone, id)
 	if err != nil {
+		if errors.IsNotFound(err) || IsNotFound(err) {
```

Comment:
These would be simplified if you make your new `IsNotFound` function handle `errors.NotFound` errors as well.